data_structures/hash_table/left_join.py

```diff
@@ -0,0 +1,13 @@
+"""This function will left join two hash maps into one."""
+
+
+def left_join(HM1, HM2):
+    """Left join 2 hash maps."""
+    final = dict()
+    for key in HM1:
+        final[key] = [HM1.get(key), None]
+    for key in HM2:
+        if key in final:
+            final[key][1] = HM2.get(key)
```

Comment:
Instead of iterating though the entire second `HashTable`, which might not have any matching keys with the first table, it would be more efficient to do this step as iterating through the first table.